File: MusicPlayer/app/src/main/java/com/example/vladimir/musicplayer/AppListDiffCallback.java
package com.example.vladimir.musicplayer;

import android.support.v7.util.DiffUtil;

public class AppListDiffCallback extends DiffUtil.ItemCallback<Track> {

    @Override
    public boolean areItemsTheSame(Track oldItem, Track newItem) {
        return oldItem.getId() == newItem.getId();
    }

    @Override
    public boolean areContentsTheSame(Track oldItem, Track newItem) {
        return oldItem.getName().equals(newItem.getName());
    }

}

File: MusicPlayer/app/src/main/java/com/example/vladimir/musicplayer/AppListDiffCallback.kt
package com.example.vladimir.musicplayer

import android.support.v7.util.DiffUtil

class AppListDiffCallback : DiffUtil.ItemCallback<Track>() {

    override fun areItemsTheSame(oldItem: Track, newItem: Track) = oldItem.id == newItem.id

    override fun areContentsTheSame(oldItem: Track, newItem: Track) = oldItem.name == newItem.name
}
